types/serde.go

```diff
@@ -16,3 +16,24 @@ func (d *Destination) UnmarshalText(text []byte) error {
 	*d = Destination(common.HexToHash(string(text)))
 	return nil
 }
+
+type JsonRpcError struct {
+	Code    int64       `json:"code"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data"`
+	Id      uint64      `json:"id"`
+}
+
+func (e JsonRpcError) Error() string {
+	return e.Message
+}
+
+var (
+	ParseError                       = JsonRpcError{Code: -32700, Message: "Parse error"}
+	InvalidRequestError              = JsonRpcError{Code: -32600, Message: "Invalid Request"}
+	MethodNotFoundError              = JsonRpcError{Code: -32601, Message: "Method not found"}
+	InvalidParamsError               = JsonRpcError{Code: -32602, Message: "Invalid params"}
+	InternalServerError              = JsonRpcError{Code: -32603, Message: "Internal error"}
+	UnexpectedRequestUnmarshalError  = JsonRpcError{Code: -32010, Message: "Unexpected unmarshal error"}
+	UnexpectedRequestUnmarshalError2 = JsonRpcError{Code: -32009, Message: "Unexpected unmarshal error"}
```

Comment:
I'm not sure I get why we want two errors like this?

Reply:
I'm not sure what is the difference between the two `UnexpectedRequestUnmarshalError` error codes. Perhaps @kerzhner could provide some additional context and advise if we should keep both or reduce to one

Reply:
I spoke with @kerzhner and updated the `Message` for those two errors to distinguish between the two